forbiditerative/plan.py

```diff
@@ -269,8 +271,33 @@ def validate_input(args):
         logging.error("Cannot use --clean-local-folder with --keep-intermediate-tasks")
         exit(1)
 
+def set_default_build_path():
+    """ 
+    Operates directly on sys.argv and sets the default value of --build. Prioritizes builds in this order:
+    1) Any provided --build path
+    2) <repo>/builds/BUILD/bin if it exists
+    3) <site-packages>/forbiditeritive/builds/BUILD/bin if it exists (the package has been installed as a library)
+
+    NOTE: Operating directly on sys.argv is generally considered bad practice but we need to modify the build arg
+    before it reaches the upstream driver code, and are unable to pass in a copy to driver. 
+    """
+    if "--build" in sys.argv:
+        #A build path has been explicitly provided by the end user, don't modify anything
+        return 
+    
+    forbiditerative_path = Path(__file__).parent 
+    regular_build_path = forbiditerative_path.parent / 'builds' / 'release' / 'bin'    
+    if os.path.exists(regular_build_path):
+        #The driver will look here by default, don't modify argv 
+        return  
+    
+    package_build_path = forbiditerative_path / 'builds' / 'release' / 'bin'    
+    if os.path.exists(package_build_path):
+        sys.argv.append("--build")
+        sys.argv.append(str(package_build_path))
```

Comment:
If none of these exist, it would be better to provide some information about various possible ways of building.

Reply:
Currently, I've left this to be handled by the upstream code in the driver, which will alert to run `build.py` if the binaries aren't found at the `--build` path or at the expected `<repo>/builds/release/bin` path.

https://github.com/IBM/forbiditerative/blob/ad16244c0ee68fd0b1228f60314fca309e2dd848/driver/run_components.py#L25-L47

It would be possible to add some additional checks in the event they aren't found and print some more relevant information on our end such as:
> the path specified by `--build` does not exist

or 
> No binaries were found at the expected paths: {{PATHS}}, please run `.\build.py`" or specify a location for the binaries using `--build`.

Reply:
I was thinking more in the direction of describing the options that correspond to these various locations: 
"If you are building locally from source, run `build.py` in the main folder.
If ...
" 
I am not sure why there are three possible locations and not two..

Reply:
How does this sound? 
```
No binaries were found in the default search locations {Repo_builds_bin_path} or {Site_package_builds_bin_path}:
If you are building locally from source, run build.py in the main folder.
If you are installing as a python package from local source, run "pip install ." in the main folder or if installing from the remote source "pip install git+https://github.com/IBM/forbiditerative.git" 
If you have manually installed the binaries in a separate location please specify a path to them with the --build argument.
NOTE: when installing locally with "pip install -e .", a --build flag must be provided to the binaries.  
```

Can you clarify which two possible locations you were expecting and why you weren't expecting the 3rd one?